connectors/uplynk/src/main/java/com/theoplayer/android/connector/uplynk/UplynkSsaiDescription.kt

```diff
@@ -71,6 +71,10 @@ data class UplynkSsaiDescription(
     val pingConfiguration: UplynkPingConfiguration = UplynkPingConfiguration()
 ) : CustomSsaiDescription() {
 
+    init {
+        check(externalIds.isEmpty() || userId != null) { "userId must be provided with external Ids" }
```

Comment:
Use `require` instead of `check`.

* `require` is for user input, and fires `IllegalArgumentException` when the check fails.
* `check` is for internal state, and fires an `IllegalStateException` when the check fails.